File: 1272-95-2161-partition-array-according-to-given-pivot/1272-95-2161-partition-array-according-to-given-pivot.js
/**
 * @param {number[]} nums
 * @param {number} pivot
 * @return {number[]}
 */
var pivotArray = function(nums, pivot) {
    const ans = new Array(nums.length).fill(0);
        let less_i = 0;
        let greater_i = nums.length - 1;
        
        for (let i = 0; i < nums.length; i++) {
            if (nums[i] < pivot) {
                ans[less_i] = nums[i];
                less_i++;
            }
        }
    
        let greater_start = nums.length - 1;
        
        for (let j = nums.length - 1; j >= 0; j--) {
            if (nums[j] > pivot) {
                ans[greater_start] = nums[j];
                greater_start--;
            }
        }
        
        while (less_i <= greater_start) {
            ans[less_i] = pivot;
            less_i++;
        }
        
        return ans;
};

File: 1272-95-2161-partition-array-according-to-given-pivot/1272-95-2161-partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int[] ans = new int[nums.length];
        int less_i = 0;
        int greater_i = nums.length - 1;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] < pivot) {
                ans[less_i] = nums[i];
                less_i++;
            }
        }

        for (int j = nums.length - 1; j >= 0; j--) {
            if (nums[j] > pivot) {
                ans[greater_i] = nums[j];
                greater_i--;
            }
        }

        while (less_i <= greater_i) {
            ans[less_i] = pivot;
            less_i++;
        }

        return ans;
    }
}